packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/blocs/pin_code_bloc.dart

```diff
@@ -78,45 +48,69 @@ class PinCodeBloc extends $PinCodeBloc {
-    isAuthenticatedWithBiometrics.connect().addTo(_compositeSubscription);
-    isPinCodeVerified.connect().addTo(_compositeSubscription);
-    availableBiometrics.connect().addTo(_compositeSubscription);
-    pinLength.connect().addTo(_compositeSubscription);
+    authenticated.connect().addTo(_compositeSubscription);
+    showBiometricsButton.connect().addTo(_compositeSubscription);
   }
 
   final PinBiometricsService biometricAuthenticationService;
   final PinCodeService pinCodeService;
   final String enterPinWithBiometrics;
 
+  BehaviorSubject<String> pinCode = BehaviorSubject.seeded('');
+  BehaviorSubject<bool> pinAuth = BehaviorSubject.seeded(false);
+  Future<bool> encryptAndVerify(String pinCode) async {
+    final String encryptedPin = await pinCodeService.encryptPinCode(pinCode);
+    final bool verifiedPin = await pinCodeService.verifyPinCode(encryptedPin);
+    return verifiedPin;
+  }
+
   @override
-  ConnectableStream<bool> _mapToIsPinCodeInSecureStorageState() =>
-      _$checkPinCodeInStorageEvent
-          .switchMap((value) =>
-              pinCodeService.isPinCodeInSecureStorage().asResultStream())
-          .setResultStateHandler(this)
-          .whereSuccess()
-          .publishReplay(maxSize: 1);
+  Stream<int> _mapToDigitsCountState() => Rx.merge([
+        _$addDigitEvent.asyncMap((digit) async {
+          final pinLength = await pinCodeService.getPinLength();
+          if (pinCode.value.length < pinLength) {
+            pinCode.add(pinCode.value += digit);
+          }
+          return pinCode.value.length;
+        }),
+        _$deleteDigitEvent.switchMap(
+          (_) {
+            pinCode.add(pinCode.value.substring(
+                0, pinCode.value.isNotEmpty ? pinCode.value.length - 1 : 0));
+            return Stream.value(pinCode.value.length);
+          },
+        ),
+      ]).startWith(0).share();
 
   @override
-  Stream<bool> _mapToAreBiometricsEnabledState() =>
-      _$checkBiometricsEnabledEvent
-          .switchMap((value) => getAreBiometricsEnabled().asResultStream())
-          .setResultStateHandler(this)
-          .whereSuccess();
+  Stream<bool> _mapToIsLoadingState() => loadingState;
+
+  @override
+  Stream<ErrorModel> _mapToErrorsState() => errorState.mapToErrorModel();
+
+  @override
+  ConnectableStream<void> _mapToAuthenticatedState() => Rx.merge([
+        _$biometricsButtonPressedEvent.asyncMap((localizedReason) async {
+          final result = await biometricAuthenticationService.enableBiometrics(
+              true, localizedReason);
+
+          if (result != null && result != BiometricsMessage.enabled) {
+            throw ErrorEnableBiometrics(result);
+          }
+          await getAreBiometricsEnabled();
+        }),
+        pinAuth.asResultStream(),
+      ]).asResultStream().setResultStateHandler(this).whereSuccess().publish();
 
   Future<bool> getAreBiometricsEnabled() async {
-    var isDeviceSupported =
+    final bool isDeviceSupported =
```

Comment:
In this method also the variables could be written only with final  without the type , like: `final isDeviceSupported `